model/task_queue_service_dependency.go

```diff
@@ -574,3 +550,27 @@ func (d *basicCachedDAGDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *
 
 	return nil
 }
+
+func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.Task) error {
+	catcher := grip.NewBasicCatcher()
+
+	blockingTasks, err := t.RefreshBlockedDependencies(dependencyCaches)
+	catcher.Add(errors.Wrap(err, "can't get blocking tasks"))
+	if err == nil && len(blockingTasks) > 0 {
+		for _, tId := range blockingTasks {
+			tDb, err := task.FindOneId(tId)
+			catcher.Add(errors.Wrapf(err, "can't get task '%s'", tId))
+			if err == nil {
+				catcher.Add(errors.Wrapf(UpdateBlockedDependencies(tDb), "can't update blocked dependencies for '%s'", tId))
+			}
+		}
+	}
+
+	blockingDeactivatedTasks, err := t.BlockedOnDeactivatedDependency(dependencyCaches)
+	catcher.Add(errors.Wrap(err, "can't get blocked status"))
+	if err == nil && len(blockingTasks) > 0 {
```

Comment:
did you mean to check the length of blockingDeactivatedTasks?

Reply:
yes 😞

Reply:
done